### ISSUE TITLE
File watcher

### ISSUE THREAD

**Issue by yonaskolb** [yonaskolb/XcodeGen#8]
An optional file watcher that automatically regenerates the project when file system changes. Should be able to specify watched files with a glob

**Comment by pepibumur** [comment]
I think you could use this for that purpose https://github.com/facebook/watchman. But thinking about it, don't you think it'd be quite hard to work with the project?

**Comment by yonaskolb** [comment]
Yes, if it kept re-gening the project from under you when you make any changes in xcode, that would be annoying.
It would have to be smart, so that only changes you make to the file system and NOT in xcode would generate the project. It would do this by watching the file system and the project at the same time.

What would probably be more useful is to simply watch the spec file for changes and regen on changes. That could use a simpler file watching dependency like https://github.com/krzysztofzablocki/KZFileWatchers which is what I use in a new open source project https://github.com/yonaskolb/Stylist

**Comment by LinusU** [comment]
I'm currently having problems when adding new files when working. My Makefile cannot (currently) pick up new files being added, and thus doesn't trigger a new generation of the Xcode project.

I would love it if XcodeGen would be able to detect new files in the source directories, and then just regenerate the project automatically 😍
